agent/ecscni/namespace_helper_windows.go

```diff
@@ -74,6 +76,26 @@ func (nsHelper *helper) ConfigureTaskNamespaceRouting(ctx context.Context, confi
 	credentialsAddressRouteAdditionCmd := fmt.Sprintf(ecsBridgeRouteAddCmdFormat, credentialsEndpointRoute, ecsBridgeEndpointName)
 	commands := []string{defaultRouteDeletionCmd, defaultSubnetRouteDeletionCmd, credentialsAddressRouteAdditionCmd}
 
+	if !config.BlockInstanceMetadata {
+		// This naming convention is drawn from the way CNI plugin names the endpoints.
+		taskPrimaryEndpointId := strings.Replace(strings.ToLower(taskENI.MacAddress), ":", "", -1)
+		taskPrimaryEndpointName := fmt.Sprintf(taskPrimaryEndpointNameFormat, taskPrimaryEndpointId, config.ContainerID)
+		imdsRouteAdditionCmd := fmt.Sprintf(ecsBridgeRouteAddCmdFormat, imdsEndpointIPAddress, taskPrimaryEndpointName)
+		commands = append(commands, imdsRouteAdditionCmd)
+	}
+
+	// Add any additional route which needs to be routed via ecs-bridge.
+	if len(config.AdditionalLocalRoutes) != 0 {
```

Comment:
nit: this check is not needed, `range` is smart :)

Reply:
+1